### PR TITLE
Add support for CRD upgrade with Helm pre-upgrade hook

### PR DESCRIPTION
**Pull request by ykulazhenkov**
This MR contains two changes:
1. Add CRDs for network-operator and sriov-network-operator to network-operator image
2. Add helm pre-upgrade hook to automatically apply CRDs

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -29,6 +31,18 @@ COPY api/ api/
 COPY controllers/ controllers/
 COPY pkg/ pkg/
 
+# Add kubectl tool
+RUN curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/${TARGETARCH}/kubectl"
+RUN chmod +x ./kubectl
+
+# Add helm chart DIR to get CRDs from it
+COPY deployment/network-operator chart
+# Update helm dependencies and copy CRDs from them
```
Comment (adrianchiris) [inline Dockerfile:40]: nit: there is no update of helm dependencies here

File: Dockerfile [hunk 2]
```diff
@@ -29,6 +31,18 @@ COPY api/ api/
...
+RUN mkdir crds && \
+    cd ./chart && \
+    cp -r ./crds /workspace/crds/network-operator/ && \
+    cp -r charts/sriov-network-operator/crds /workspace/crds/sriov-network-operator/
```
Comment (adrianchiris) [inline Dockerfile:44]: do we want NFD CRDs ? or we going to have issues with NFD deployed from GPU Op ?

File: deployment/network-operator/README.md [hunk 3]
```diff
@@ -183,22 +183,6 @@ helm search repo mellanox/network-operator -l
 
 > __NOTE__: add `--devel` option if you want to list beta releases as well
 
-### Download CRDs for the specific release
```
Comment (adrianchiris) [inline deployment/network-operator/README.md:186]: Im thinking we should still keep a section about upgrading CRDs.

we should have two subsections

1. manual upgrade (what we have here)
2. automatic upgrade using helm chart by setting .Values.upgradeCRDs to true

Comment (ykulazhenkov) [reply deployment/network-operator/README.md:186]: I changed this section to keep info about manual process

File: Dockerfile (hunk 1 above)
Comment (ykulazhenkov) [reply Dockerfile]: Correct. I updated the docstring. Update here will be required after we will upgrade the version of the NFD chart

**Review by adrianchiris — APPROVED** [review]

**Review by rollandf — APPROVED** [review]

**Comment by ykulazhenkov** [comment]
Latest released kubectl has security known security issue which is detected by image_scan job. Need to wait for new kubectl release (the issue is already fixed in master)

**Pull request merged**